src/components/DataTable/index.tsx

```diff
@@ -49,41 +49,15 @@ export function DataTable<TableColumns, Data extends TableColumns>({
   groupBy,
   hiddenColumns,
   className,
-  usageSection,
   isVirtualized = false,
   virtualizedRowEstimatedHeight = 35,
+  sortBy = [],
   ...rest
 }: DataTableProps<TableColumns, Data>) {
-  const [queryParams, setQueryParams] = useSearchParams({
-    sortBy: "",
-    sortOrder: ""
-  });
-
-  const sortField = queryParams.get("sortBy");
-
-  const isSortOrderDesc =
-    queryParams.get("sortOrder") === "desc" ? true : false;
+  const [queryParams, setQueryParams] = useSearchParams();
 
   const tableContainerRef = useRef<HTMLDivElement>(null);
 
-  const sortBy = useMemo(() => {
-    const data = sortField
-      ? [
-          {
-            id: sortField,
-            desc: isSortOrderDesc
-          }
-        ]
-      : [];
-    if (sortField === "config_type" && usageSection === "config-list") {
```

Comment:
We need this to sort the config list by config_type & name whenever user tries sorting by config_type column. I see this logic is being removed in the refactor. could you please handle this.

Reply:
I am lifting this to ConfigList, and then it passed it in via as the DataTable shouldn't care about specific implementation details. Any consumer of DataTable should now pass in SortBy, which is compatible with react-table, ensuring we reduce the no. of implementation details in the DataTable component. One question though. Do we want to sort by config type and then name?

Reply:
Moving this to ConfigList is fine. But we need to handle sorting by confg_type + name whenever user tries sorting by config_type

Reply:
In that case, we can expose setSort or something that updates the sort state whenever it changes.

Reply:
I think moving lines from 78 to 85 here https://github.com/flanksource/flanksource-ui/pull/661/files#diff-fe40532367ba1cce6d73ad3ef76bfdbdbfebc7fd4b865afc5f704ebf3689c3b2R459 will handle the use case.